scripts/generate/genesis.sh

```diff
@@ -32,7 +32,7 @@ F=100 # fake avvm keys
 # print commit
 git show HEAD --oneline | tee $outputDir/genesisCreation.log
 
-keygenCmd="stack exec cardano-keygen -- --genesis-dir $outputDir -m $M -n $N --richmen-share 0.94 --testnet-stake 19072918462000000 --utxo-file $utxo_file --blacklisted $blacklisted --fake-avvm-entries $F"
+keygenCmd="stack exec cardano-keygen -- generate-genesis --genesis-dir $outputDir -m $M -n $N --richmen-share 0.94 --testnet-stake 19072918462000000 --utxo-file $utxo_file --blacklisted $blacklisted --fake-avvm-entries $F"
```

Comment:
> stack exec -- cardano-keygen OPTS 

AFAIK the order doesn't matter.

> Why generate-genesis without --?

Probably because it's a command, not option (haven't looked at the code yet, though).